### PR TITLE
Option to specify custom phantomjs launcher file

### PR DESCRIPTION
**Pull request by dhiemstra**
Also discussed in #876 sometimes you want to be able to override the phantomjs viewPortSize option. As far as I know this is not possible using command line arguments and the `browser_args` option.

If there is any better option than this please let me know ;-)

### CODE REVIEW

**Comment by dhiemstra** [comment]
Just found out how to specify this via browser_args in #398 :neckbeard:

File: docs/config_file.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ Common Configuration Options
     report_file:                 [String]  file to write test results to (stdout)
     xunit_intermediate_output    [Boolean] print tap output for the xunit reporter (false)
     phantomjs_debug_port:        [Number]  port used to attach phantomjs debugger
+    phantomjs_launcher:          [String]  path of custom phantomjs launcher file
```
Comment (johanneswuerbach) [inline docs/config_file.md]: I would prefer `phantomjs_script`.

File: lib/utils/known-browsers.js [hunk 2]
```diff
@@ -137,7 +137,12 @@ function knownBrowsers(platform, config) {
       name: 'PhantomJS',
       possibleExe: 'phantomjs',
       args: function() {
-        var options = [path.resolve(__dirname, '../../assets/phantom.js'), this.getUrl()];
+        var launcher = config.get('phantomjs_launcher');
+        if (!launcher) {
+          launcher = path.resolve(__dirname, '../../assets/phantom.js');
+        }
+        console.log(launcher);
```
Comment (johanneswuerbach) [inline lib/utils/known-browsers.js]: Looks like your forgot a console.log

**Comment by johanneswuerbach** [comment]
> Just found out how to specify this via browser_args in #398 :neckbeard:

Could you add this maybe to the README?

**Comment by dhiemstra** [comment]
Actually, I couldn't make that work. Can't think of any other way than customizing the `phantomjs_script`.

**Comment by johanneswuerbach** [comment]
Looks great so far, could you add a test here https://github.com/testem/testem/blob/master/tests/utils/known-browsers_tests.js#L370 to ensure we don't regress?

**Comment by dhiemstra** [comment]
Sure thing, here you go @johanneswuerbach

**Comment by johanneswuerbach** [comment]
Thanks 👏

**Pull request merged**